datalad/support/gitrepo.py

```diff
@@ -1577,18 +1583,18 @@ def get_hexsha(self, commitish=None, short=False):
             return stdout[0]
 
     @normalize_paths(match_return_type=False)
-    def get_last_commit_hash(self, files):
+    def get_last_commit_hexsha(self, files):
         """Return the hash of the last commit the modified any of the given
         paths"""
         try:
-            stdout, stderr = self._git_custom_command(
-                files,
-                ['git', 'log', '-n', '1', '--pretty=format:%H'],
-                expect_fail=True)
-            commit = stdout.strip()
-            return commit
+            commit = self.call_git(
+                ['rev-list', '-n1', 'HEAD'],
+                files=files,
+            )
+            commit = commit.strip()
+            return commit if commit else None
         except CommandError as e:
-            if 'does not have any commits' in e.stderr:
+            if 'unknown revision or path not in the working tree' in e.stderr:
```

Comment:
From my POV it'd be best to avoid brittle error message matching if we can.  Given we're in the error handling arm, another git call seems acceptable, so this could be replaced by `if self.get_hexsha() is None`.  (`get_hexsha()` relies on `format_commit` which itself does error message matching, but at least we'd avoid introducing another spot with a new message.)

Reply:
Done.